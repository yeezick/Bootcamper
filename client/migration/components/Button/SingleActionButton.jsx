import { useState } from 'react';
import { Text, TouchableHighlight } from 'react-native';
import { getButtonStyles } from './styles';

export const SingleActionButton = (props) => {
  const [pressed, togglePressed] = useState(false);
  const { handler, title } = props.payload;
  const { style, type } = props;
  const styles = getButtonStyles(style, type);

  const touchProps = {
    onPress: handler,
    style: { ...styles.container, borderWidth: pressed ? 0 : styles.container.borderWidth },
    underlayColor: '#999999',
    onShowUnderlay() {
      togglePressed(true);
    },
    onHideUnderlay() {
      togglePressed(false);
    },
  };

  return (
    <TouchableHighlight {...touchProps} disabled={style === 'disabled' && true}>
      <Text style={styles.text}>{title}</Text>
    </TouchableHighlight>
  );
};
