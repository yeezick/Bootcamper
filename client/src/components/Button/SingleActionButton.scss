.single-button {
  align-items: center;
  border: none;
  border-radius: 10px;
  display: flex;
  font-size: 16px;
  font-weight: 'Nunito';
  height: 50px;
  justify-content: center;
  margin-bottom: 10px;
  text-align: center;
  width: 102px;
}
    
.default {
  background-color: #1b1b1b;
  color: #fff;
}

.disabled {
  background-color: #ebebe4;
  color: #5c5c5c
}

.light {
  background-color: rgba(0,0,0,0);
  border: 1px solid #000;
  color: #000;
}

.long {
  width: 250px;
}