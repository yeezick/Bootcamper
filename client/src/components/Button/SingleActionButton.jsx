import './SingleActionButton.scss'

export const SingleActionButton = ({text, onClick, type}) => {
    return (
        <div className="single-button">
            <button onClick={onClick} type={type} >{text}</button>
        </div>

    )
}