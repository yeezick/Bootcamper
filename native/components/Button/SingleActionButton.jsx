import { Button } from "react-native";
import "./SingleActionButton.scss";

export const SingleActionButton = ({
  text,
  onClick = null,
  type = "button",
}) => {
  return (
    // <div className="single-button">
    <Button onPress={onClick} type={type} title={text} />
  );
};
