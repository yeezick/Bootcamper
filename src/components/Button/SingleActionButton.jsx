import './SingleActionButton.scss';

/**
 * @name SingleActionButton
 * @param {Object} payload - Contains the handler and button text
 * @param {string} style -  Decides which color scheme (disabled/light) should be applied. If empty, default style is applied.
 * @param {string} size - Decides whether the width of the button (long). If empty, button is short.
 */
export const SingleActionButton = (props) => {
  console.log(props.payload)
  const { title } = props.payload;
  const { style, size } = props;
  const isDisabled = style === 'disabled' ? true : false;

  return props.payload.handler ? (
    <button
      className={`single-button ${style} ${size ? size : ''}`}
      disabled={isDisabled}
      onClick={props.payload.handler}
    >
      {title}
    </button>
  ) : (
    <button
    className={`single-button ${style} ${size ? size : ''}`}
    disabled={isDisabled}
  >
    {title}
  </button>);
};
