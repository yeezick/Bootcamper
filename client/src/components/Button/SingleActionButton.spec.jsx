// import {render, screen} from "@testing-library/react"
// import { SingleActionButton } from "./SingleActionButton"
// import '@testing-library/jest-dom'

// describe('sample test to ensure tests are passing', () => {
//     test('renders button', () => {
//         render(<SingleActionButton />)
//         const buttonElement = screen.getByTestId("button");
//         expect(buttonElement).toBeInTheDocument()
//     })
// })


import React from 'react';
import { SingleActionButton } from "./SingleActionButton"
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';


/*
*TEST PLAN:
-renders successfully
-passsing in a string to the title prop should also be rendered
-passing in a handler to the button will successfully execute it on press
-if a style prop is "disabled", then background should = '#EBEBE4'and text should be '#5c5c5c'
-if a style prop doesnt exist ("defaultStyle"), then background should = '#1b1b1b' and text should be '#fff'
-if a style prop is "lightStyle", then background should = 'rgba(0,0,0,0)', borderWidth: 1, borderColor: '#000',and text should be '#000'
-if type="long", check if width is set to 250
 */

it('renders', () => {
  const { asFragment } = render(<SingleActionButton />);
  expect(asFragment()).toMatchSnapshot();

});

test('It has to contain My Button', () => {
  render(
    <div>
      <SingleActionButton />
    </div>
  )

  const buttonClass = SingleActionButton().type.styledComponentId
  const MyButtonRoots = document.getElementsByClassName(buttonClass)
  console.log(MyButtonRoots)
  const style = window.getComputedStyle(MyButtonRoots[0])
  
  expect(style.position).toBe('fixed')
  expect(style.top).toBe('0px')
})

