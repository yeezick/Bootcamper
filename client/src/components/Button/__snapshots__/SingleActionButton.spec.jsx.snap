// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="single-button"
  >
    <button
      data-testid="button"
      type="button"
    />
  </div>
</DocumentFragment>
`;
